src/Compilers/Core/Portable/WellKnownTypes.cs

```diff
@@ -662,6 +665,9 @@ internal static class WellKnownTypes
             "System.MissingMethodException",
             "System.Runtime.CompilerServices.MetadataUpdateOriginalTypeAttribute",
             "System.Runtime.CompilerServices.Unsafe",
+
+            "System.Threading.Lock",
+            "System.Threading.Lock+Scope",
```

Comment:
>System.Threading.Lock+Scope [](http://example.com/codeflow?start=13&length=27)

Is this the first time we dealing with a nested well known type? Doe the implementation guarantee that it looks for the nested type within what is considered a ```WellKnownType.System_Threading_Lock```? We might need to adjust the implementation to explicitly enforce that.